### PR TITLE
Add tests reflecting #19390

### LINKED ISSUE #17734: Test passing though the code doesn't check all test cases.
#### Describe your problem and - if possible - how to reproduce it
I am using this code and it is passing though I am not checking it for all users. 
while(users.hasOwnProperty("Alan")){
    return true;
  }
  return false; 

#### Add a Link to the page with the problem
https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/basic-data-structures/check-if-an-object-has-a-property

#### Tell us about your browser and operating system

* Browser Name: Chrome
* Browser Version: Version 67.0.3396.99 (Official Build) (64-bit)
* Operating System: Windows 10

### LINKED ISSUE #19382: [Bug] Basic Data Structures: Check if an Object has a Property
**Describe your problem and how to reproduce it:**
Found from this PR https://github.com/freeCodeCamp/freeCodeCamp/pull/19367
The tests are only checking if all the properties exists and another check is done after deleting all the properties so code like:
```js
function isEveryoneHere(obj) {
  return users.hasOwnProperty('Alan', 'Jeff', 'Sarah', 'Ryan'); //shouldnt pass since .hasOwnProperty() doesnt take more than 1 parameter but would only check the first one.
}
```
and 
```js
function isEveryoneHere(obj) {
  return obj.hasOwnProperty("Alan");
}
```
Passes the challenge whereas they shouldn't.

This can be resolved by adding checks after deleting each property or randomly deleting one and then checking.
**Add a Link to the page with the problem:**
https://learn.freecodecamp.org/javascript-algorithms-and-data-structures/basic-data-structures/check-if-an-object-has-a-property/

**Tell us about your browser and operating system:**
* Browser Name: Chrome 
* Browser Version: 69
* Operating System: 10

### PR DESCRIPTION
**Pull request by Lewis65**
Added two tests to prevent checking a single property and passing the challenge.

The tests check two names at a time rather than adding 4 individual tests to the challenge.

- [x] I have read [freeCodeCamp's contribution guidelines](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/CONTRIBUTING.md).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] None of my changes are plagiarized from another source without proper attribution.
- [x] My article does not contain shortened URLs or affiliate links.

If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.

Closes #19382
Closes #17734

### CODE REVIEW

**Comment by Lewis65** [comment]
First PR, so if I am doing something wrong please let me know :)

**Comment by ezioda004** [comment]
Hi @Lewis65,
Waiting for the conformation from the core team to accept PR for that bug.
Meanwhile I've notice you've introduced 2 tests, though it can still be passed by doing something like:
```js
return obj.hasOwnProperty('Alan') && obj.hasOwnProperty('Jeff');
```
I'd suggest adding tests for individual Properties, that way all would need to be present.

File: curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/check-if-an-object-has-a-property.english.md [hunk 1]
```diff
@@ -26,7 +26,10 @@ tests:
     testString: 'assert(isEveryoneHere(users) === true, "The function <code>isEveryoneHere</code> returns <code>true</code> if <code>Alan</code>, <code>Jeff</code>, <code>Sarah</code>, and <code>Ryan</code> are properties on the <code>users</code> object");'
   - text: 'The function <code>isEveryoneHere</code> returns <code>false</code> if <code>Alan</code>, <code>Jeff</code>, <code>Sarah</code>, and <code>Ryan</code> are not properties on the <code>users</code> object'
     testString: 'assert((function() { delete users.Alan; delete users.Jeff; delete users.Sarah; delete users.Ryan; return isEveryoneHere(users) })() === false, "The function <code>isEveryoneHere</code> returns <code>false</code> if <code>Alan</code>, <code>Jeff</code>, <code>Sarah</code>, and <code>Ryan</code> are not properties on the <code>users</code> object");'
-
+    - text: 'The function <code>isEveryoneHere</code> returns <code>false</code> if <code>Alan</code>, <code>Jeff</code>, <code>Sarah</code>, and <code>Ryan</code> are not properties on the <code>users</code> object'
```
Comment (ezioda004) [inline curriculum/challenges/english/02-javascript-algorithms-and-data-structures/basic-data-structures/check-if-an-object-has-a-property.english.md]: Can you also make sure the indentation of the `text` is same as above?

**Comment by nikrb** [comment]
@ezioda004 tested locally. 
confidence is low, but I don't think this challenge is working yet as it passes with just `return 'Alan' in obj;` as the solution. I think the problem is that the new tests remove users but don't add them back in, so Alan is always missing.
Apologies if I'm being too pedantic or have missed something.

**Comment by ezioda004** [comment]
Hi @nikrb,

Thanks for QA'ing. You are correct, after deleting `users.Alan`, its never added back and the subsequent tests returns `false`.

@Lewis65 Could you add the keys back in the next tests?
Example:
```js
(function() { delete users.Jeff; users.Alan = {age: 27, online: true}; return isEveryoneHere(users) })() === false
```
Likewise for other tests.

**Comment by ValeraS** [comment]
@ezioda004 @nikrb For now, JS challenges execute tests independently. No need to restore initial data anymore.

**Comment by nikrb** [comment]
tested this locally on latest master. curriculum test passes and client app looks good.

**Comment by RandellDawson** [comment]
@ezioda004 Can you review this PR again to see if everything meets the requirement to close issue #19382?

**Review by RandellDawson — APPROVED** [review]
I have tested this PR locally and conclude it now fixed the bug described in the two issues #17734 and #19382. Great job @Lewis65!

**Review by ezioda004 — APPROVED** [review]

**Comment by ezioda004** [comment]
Thank you @Lewis65 for your contribution and congratulations on your first PR merge to this repo! 🎉
Hope to see more contributions from you in the future.

**Pull request merged**